lasagne/layers/input.py

```diff
@@ -49,6 +49,12 @@ class InputLayer(Layer):
     """
     def __init__(self, shape, input_var=None, name=None, **kwargs):
         self.shape = shape
+        if any(d is not None and d <= 0 for d in self.shape):
+            raise ValueError((
+                "Cannot create InputLayer with a non-positive shape "
+                "dimension. shape=%r, self.name=%s") % (
+                    self.shape, str(name)))
```

Comment:
`%r` also has the advantage that you can still distinguish between a layer that has no name and a layer of the name `"None"`. Not that it would matter really, but it reads better.